test/collections.js

```diff
@@ -123,6 +123,11 @@ describe('Collections', function () {
       expect(map2).to.have.entries([['a', 'b'], ['c', 'd']]);
     });
 
+    it('non-entry objects should throw', function () {
+      expect(new Map([1, 2, 3])).to['throw'](TypeError);
```

Comment:
ok - so the specific step `If Type(nextItem) is not Object` is what gets triggered here, in both cases, because strings and numbers are primitives. However, `[{}, {}, {}]` would be valid, as would `[[], [], []]`. I'll add some tests, and then patch them.